libs/clients/regulations-admin/src/lib/apiConfiguration.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  Configuration,
+  DraftAuthorApi,
+  DraftRegulationCancelApi,
+  DraftRegulationChangeApi,
+  DraftRegulationsApi,
+} from '../../gen/fetch'
+import { Provider } from '@nestjs/common/interfaces/modules/provider.interface'
+
+import { RegulationsAdminClientConfig } from './RegulationsAdminClientConfig'
+import { ConfigType } from '@nestjs/config'
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+
+const provideApi = <T>(
+  Api: new (configuration: Configuration) => T,
+): Provider<T> => ({
+  provide: Api,
+  useFactory: (config: ConfigType<typeof RegulationsAdminClientConfig>) => {
+    return new Api(
+      new Configuration({
+        fetchApi: createEnhancedFetch({
+          name: 'Regulations-AdminClientService',
+          logErrorResponseBody: true,
+        }),
+        basePath: config.baseApiUrl,
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+      }),
+    )
+  },
+  inject: [RegulationsAdminClientConfig.KEY],
+})
+
+export const DraftAuthorApiProvider = provideApi(DraftAuthorApi)
+export const DraftRegulationCancelApiProvider = provideApi(
+  DraftRegulationCancelApi,
+)
+export const DraftRegulationChangeApiProvider = provideApi(
+  DraftRegulationChangeApi,
+)
+export const DraftRegulationsApiProvider = provideApi(DraftRegulationsApi)
```

Comment:
This can be simplified by doing

```
export const exportedApis = [DraftAuthorApi, DraftRegulationCancelApi, DraftRegulationChangeApi, DraftRegulationsApi].map(
  (Api) => ({
    provide: Api,
    useFactory: (configuration: Configuration) => new Api(configuration),
    inject: [ApiConfiguration.provide],
  }),
)
```

Where the configuration is created as a provider

```
export const ApiConfiguration = {
  provide: 'RegulationsAdminClientConfiguration',
  useFactory: (config: ConfigType<typeof RegulationsAdminClientConfig>) => {
    return new Configuration({
      fetchApi: createEnhancedFetch({
        name: 'clients-regulations-admin-api',
      }),
      basePath: config.baseApiUrl,
       headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
    })
  },
  inject: [RegulationsAdminClientConfig.KEY],
}
```
You can see an example here: https://github.com/island-is/island.is/blob/main/libs/clients/auth/delegation-api/src/lib/apis.ts